### DISCUSSION TITLE
first time make ender 3 v2 firmware need help

### DISCUSSION THREAD

**Discussion by ender3Blake** [Jyers/Marlin#1560] [Q&A]
I am upgrading my ender 3 v2 to a new hotend so i can print at hotter temps im not sure how to make sure the firmware is set for my 4.2.2 board on my machine. ive open the source code on vs and think i have find every spot i need to change setting but just not sure how make its right for my board. Can someone help me with this to compile the firmware correctly or tel me how to do becase i would love to be able to compile my own firmware.

**Comment by gilegue** [comment]
If you just want to know how to compile firmware (assuming all parameters has been set correctly), the easiest path would be installing plug-ins needed in visual studio code. You will need to install platformio and marlin autobuild. You can find a step by step videos on YouTube as well (I will recommend to start with teaching tech channel if you just started, such as https://youtu.be/eq_ygvHF29I or https://youtu.be/J9vxJT5Tgh4. Another good starting point will be Brian's video https://youtu.be/o1zXYwgIPLY  here). It is not that hard if you follow those guide carefully.

**Reply by ender3Blake** [reply]
That’s thing I don’t know is how I know what board it’s set for and I will check these videos out. I just know I gott change the temp Thermistor setting and the home area Bc of the printed hotend holder.

**Reply by gilegue** [reply]
For 4.2.2 board, open Configuration.h and find #define MOTHERBOARD. After the space set whatever value there to BOARD_CREALITY_V4 (so the entire line will be like #define MOTHERBOARD BOARD_CREALITY_V4).

**Reply by ender3Blake** [reply]
Okay thank you for the info on how to do that

**Comment by Cal-LF** [comment]
Compiling info in #1511
